lib/buffer.js

```diff
@@ -289,6 +289,24 @@ Buffer.prototype.compare = function compare(b) {
 };
 
 
+Buffer.prototype.indexOf = function indexOf(val, byteOffset) {
+  if (byteOffset > 0xffffffff)
+    byteOffset = 0xffffffff;
+  if (byteOffset < 0)
+    byteOffset = 0;
+  byteOffset = byteOffset >>> 0;
+
+  if (typeof val === 'string')
+    return internal.indexOfString(this, val, byteOffset);
+  if (val instanceof Buffer)
+    return internal.indexOfBuffer(this, val, byteOffset);
+  if (typeof val === 'number')
+    return internal.indexOfNumber(this, val, byteOffset);
```

Comment:
Because many of them are stupid. I first wanted some of the coercion and checks that are commonly incorrectly done (e.g. `uint32` coercion) then it was decided all checks should go in. I was okay with macros being replaced to not add any additional runtime overhead, then it was decided that they'd be replaced with functions. I never wanted a bunch of function checks to do simple things like `typeof` for types that are easily checked.